src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt

```diff
@@ -24,29 +22,123 @@ class ThemePatch : ResourcePatch {
+            "drawable",
+            "drawable-anydpi",
+            "drawable-hdpi",
+            "drawable-ldrtl",
+            "drawable-ldrtl-night",
+            "drawable-ldrtl-xxhdpi",
+            "drawable-mdpi",
+            "drawable-night",
+            "drawable-nodpi",
+            "drawable-sw600dp",
+            "drawable-sw600dp-xhdpi",
+            "drawable-sw720dp-xhdpi",
+            "drawable-watch",
+            "drawable-xhdpi",
+            "drawable-xxhdpi",
+            "drawable-xxxhdpi"
         )
 
+        context.xmlEditor[attrsPath].use { editor ->
+            val file = editor.file
+
+            val resourceNode = file.getElementsByTagName("resources").item(0) as Element
+
+            resourceNode.appendChild(
+                file.createElement("attr").apply {
+                    setAttribute("format", "reference")
+                    setAttribute("name", "splashScreenColor")
+                }
+            )
+        }
+        valuesPaths.forEachIndexed { pathIndex, valuesPath ->
+            val stylesPath = "res/$valuesPath/styles.xml"
+
+            context.xmlEditor[stylesPath].use { editor ->
+                val file = editor.file
+
+                val childNodes = (file.getElementsByTagName("resources").item(0) as Element).childNodes
+
+                for (i in 0 until childNodes.length) {
+                    val node = childNodes.item(i) as? Element ?: continue
+                    val nodeAttributeName = node.getAttribute("name")
+
+                    file.createElement("item").apply {
+                        setAttribute(
+                            "name",
+                            when (pathIndex) {
+                                0 -> "splashScreenColor"
+                                1 -> "android:windowSplashScreenBackground"
+                                else -> "null"
+                            }
+                        )
+
+                        appendChild(
+                            file.createTextNode(
+                                when (pathIndex) {
+                                    0 -> when (nodeAttributeName) {
+                                        "Base.Theme.YouTube.Launcher.Dark" -> darkThemeBackgroundColor
+                                        "Base.Theme.YouTube.Launcher.Light" -> lightThemeBackgroundColor
+                                        else -> "null"
+                                    }
+                                    1 -> when (nodeAttributeName) {
+                                        "Base.Theme.YouTube.Launcher" -> "?attr/splashScreenColor"
+                                        else -> "null"
+                                    }
+                                    else -> "null"
+                                }
+                            )
+                        )
+
+                        if (this.textContent != "null")
+                            node.appendChild(this)
+                    }
```

Comment:
Please add explanatory comments on why this is required, and what purpose it serves.

Reply:
This needs to add change the value of field named ```splashScreenColor``` inside the file ```attr.xml```.

But to prevent a line from being added after ```Base.Theme.YouTube.Launcher```(check the code), I decided to add this only when a specific file is indexed by ```ForEachIndexed``` method

Reply:
> This needs to add change the value of field named splashScreenColor inside the file attr.xml.

Aside from the semantic issues I have with this sentence, you are supposed to comment on what the purpose of the code is and why it is required and not explain what the code does. We can simply read the code and see that it changes a field named `splashScreenColor`, but no comment explains the purpose or why it is required. I suggest looking up (this)[https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/].

Reply:
> > This needs to add change the value of field named splashScreenColor inside the file attr.xml.
> 
> Aside from the semantic issues I have with this sentence, you are supposed to comment on what the purpose of the code is and why it is required and not explain what the code does. We can simply read the code and see that it changes a field named `splashScreenColor`, but no comment explains the purpose or why it is required. I suggest looking up [this](https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/).

Yeah...sorry. There was a syntax error (now fixed) in the comment.

By the way ok...I'll add a comment for this too.

Reply:
Sounds good, make sure to follow the link I sent earlier, it is quite useful.

Reply:
> Sounds good, make sure to follow the link I sent earlier, it is quite useful.

I'll add a comment with the same style of ones added on [this](https://github.com/revanced/revanced-integrations/commit/9ab8a646ed07d709c46fe7b5dd3238bc23301b8b) merged code